### PR TITLE
Add support for DL Stage

### PR DESCRIPTION
**Pull request by eddybl**
Source is the internal patch file

I removed the `std::cout` commands and also therefore `#include <iostream>`

This is IBPT internal review, after that I would create a PR to upstream

### CODE REVIEW

File: newportApp/src/AG_CONEX.cpp [hunk 1]
```diff
@@ -447,12 +470,12 @@ asynStatus AG_CONEXAxis::setDGain(double dGain)
 {
   asynStatus status = asynSuccess;
   bool closedLoop;
-  //static const char *functionName = "AG_CONEXAxis::setPGain";
+  //static const char *functionName = "AG_CONEXAxis::setDGain";
```
Comment (smarsching) [inline newportApp/src/AG_CONEX.cpp]: This line is commented out, so I do not understand why this change is needed.

File: newportApp/src/AG_CONEX.cpp [hunk 2]
```diff
@@ -496,12 +519,29 @@ asynStatus AG_CONEXAxis::poll(bool *moving)
...
-  if (count != 1) goto skip;
+  
+  if (conexModel_ == ModelDL) {
+	  // The TS command returns 8 characters (1TSabcdefgh).
+	  // The first character (a) represents the status bits in Hexadecimal.
+	  // The next 5 characters (bcdef) represent the error bits in Hexadecimal.
+	  // The last two characters (gh) represent the Controller state in Hexadecimal.
+	  // we read last two hex characters, the state
+	  count = sscanf(pC_->inString_, "%*dTS%*6c%x", &status);
+	  if (count != 1) goto skip;
```
Comment (smarsching) [inline newportApp/src/AG_CONEX.cpp]: This indendation does not seem to be consistent with what the rest of the code uses. Maybe spaces and tabs accidentally got mixed?

File: newportApp/src/AG_CONEX.cpp [hunk 3]
```diff
@@ -496,12 +519,29 @@ asynStatus AG_CONEXAxis::poll(bool *moving)
...
+  } 
+  else {
```
Comment (smarsching) [inline newportApp/src/AG_CONEX.cpp]: This formatting (newline preceding `else`) does not appear to be consistent with what the rest of the code uses.

Yes, there is one place earlier in the code that uses the same style, so there already is some inconsistency in the exsting code, but the prevailing style seems to be no newline before `else`, so I suggest going with that one.

File: newportApp/src/AG_CONEX.cpp [hunk 4]
```diff
@@ -496,12 +519,29 @@ asynStatus AG_CONEXAxis::poll(bool *moving)
...
+    count = sscanf(pC_->inString_, "%*dTS%*4c%x", &status);
+    if (count != 1) goto skip;
+  }
 
-  state = status & 0xff;
-  if ((state == 0x1e) || (state == 0x28)) done = 0;
+  if (conexModel_ == ModelDL) {
+	  if (status == 0x3c) done=0;
+  }
+  else {
```
Comment (smarsching) [inline newportApp/src/AG_CONEX.cpp]: Same formatting issues as earlier.

**Review by smarsching — CHANGES_REQUESTED** [review]
In general, this looks good to me, I just found a few inconsistencies regarding formatting of the code and one change that looks redundant to me.

File: newportApp/src/AG_CONEX.cpp (hunk 1 above)
Comment (eddybl) [reply newportApp/src/AG_CONEX.cpp]: Just noticed that this is a clear copy & paste error.

Not sure why these lines even exist, but if they are sometimes used for debugging it might be tricky to figure out that there is actually a typo

Comment (eddybl) [reply newportApp/src/AG_CONEX.cpp]: but I can also leave this out or do this as a separate commit

File: newportApp/src/AG_CONEX.cpp [hunk 5]
```diff
@@ -496,12 +519,27 @@ asynStatus AG_CONEXAxis::poll(bool *moving)
...
+  } else {
+    // The response string is of the form "1TSabcdef"
+    count = sscanf(pC_->inString_, "%*dTS%*4c%x", &status);
+    if (count != 1) goto skip;
+  }
 
-  state = status & 0xff;
-  if ((state == 0x1e) || (state == 0x28)) done = 0;
+  if (conexModel_ == ModelDL) {
+	  if (status == 0x3c) done=0;
```
Comment (smarsching) [inline newportApp/src/AG_CONEX.cpp]: The indentation still needs to be fixed here.

**Review by smarsching — APPROVED** [review]
There is one line remaining where the indendation is still wrong. Apart from this, everything looks good to me now.

Your point about the typo in the debug code is valid. I agree that it is probably better to fix this in a separate commit, because it is unrelated to adding support for the DL stage.

I think that the whole handling of debugging isn’t very nice at the moment. It would be better to use asynPrint wherever possible and use conditional debug statements everywhere else (or, if someone is really worried about performance, wrap the debug code in `#ifdef`s). Just about every other solution is clearner than leaving commented out code in the sources. However, this is a completely separate issue, so I do not think that this has to or should be addressed in the context of this PR.
